benchalerts/clients.py

```diff
@@ -0,0 +1,198 @@
+
+            adapter = MockAdapter()
+        else:
+            adapter = HTTPAdapter(max_retries=retry_strategy)
+
+        self.session = requests.Session()
+        self.session.mount("https://", adapter)
+
+    def get(self, path: str) -> dict:
+        log.debug(f"GET {self.base_url}{path}")
+        res = self.session.get(url=self.base_url + path, timeout=self.timeout_s)
+        res.raise_for_status()
+        return res.json()
+
+    def post(self, path: str, json: dict) -> Optional[dict]:
+        log.debug(f"POST {self.base_url}{path} {json}")
+        res = self.session.post(
+            url=self.base_url + path, json=json, timeout=self.timeout_s
+        )
+        res.raise_for_status()
+        if res.content:
+            return res.json()
+
+
+class GithubRepoClient(_BaseClient):
+    """A client to interact with a Github repo.
+
+    Parameters
+    ----------
+    repo
+        The repo name, in the form 'owner/repo'.
+
+    Environment variables
+    ---------------------
+    GITHUB_API_TOKEN
+        A Github API token with ``repo`` access.
+    """
+
+    def __init__(self, repo: str):
+        token = os.getenv("GITHUB_API_TOKEN")
+        if not token:
+            fatal_and_log("Environment variable GITHUB_API_TOKEN not found")
+
+        super().__init__()
+        self.session.headers = {"Authorization": f"Bearer {token}"}
+        self.base_url = f"https://api.github.com/repos/{repo}"
+
+    def create_pull_request_comment(
+        self,
+        comment: str,
+        *,
+        pull_number: Optional[int] = None,
+        commit_sha: Optional[str] = None,
+    ):
+        """Create a comment on a pull request, specified either by pull request number
+        or commit SHA.
+
+        Parameters
+        ----------
+        comment
+            The comment text.
+        pull_number
+            The number of the pull request. Specify either this or ``commit_sha``.
+        commit_sha
+            The SHA of a commit associated with the pull request. Specify either this
+            or ``pull_number``.
+        """
+        if commit_sha:
+            pull_numbers = [
+                pull["number"] for pull in self.get(f"/commits/{commit_sha}/pulls")
+            ]
+            if len(pull_numbers) != 1:
+                fatal_and_log(
+                    "Need exactly 1 pull request associated with commit "
+                    f"'{commit_sha}'. Found {pull_numbers}."
+                )
+            pull_number = pull_numbers[0]
+        elif not pull_number:
+            fatal_and_log("Either specify pull_number or commit_sha")
+
+        log.info(
+            f"Posting the following message to pull request #{pull_number}:\n\n"
+            + comment
+        )
+        return self.post(f"/issues/{pull_number}/comments", json={"body": comment})
+
+
+class ConbenchClient(_BaseClient):
+    """A client to interact with a Conbench server.
+
+    Parameters
+    ----------
+    server_url
```

Comment:
Yeah I saw that other projects store this in an environment variable. I didn't think it was particularly sensitive so I decided not to. But since everyone else does, it's probably worth it to switch back and read it from the environment.

Reply:
done!